### PR TITLE
Subscription: support TsFile message format & improve exception handling logic

### PR DESCRIPTION
**Pull request by VGalaxies**
Support TsFile as a payload in data subscriptions.

Follow-up TODOs:

- [x] Notify the client when the topic does not meet the restrictions.
- [ ] Adapt the data subscription side to support the changes in #12490.

### CODE REVIEW

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/config/ConsumerConstant.java [hunk 1]
```diff
@@ -34,31 +34,34 @@ public class ConsumerConstant {
...
 
   public static final String ENDPOINTS_SYNC_INTERVAL_MS_KEY =
       "endpoints-sync-interval-ms"; // unit: ms
-  public static final long ENDPOINTS_SYNC_INTERVAL_MS_DEFAULT_VALUE = 30000;
-  public static final long ENDPOINTS_SYNC_INTERVAL_MS_MIN_VALUE = 5000;
+  public static final long ENDPOINTS_SYNC_INTERVAL_MS_DEFAULT_VALUE = 120_000;
+  public static final long ENDPOINTS_SYNC_INTERVAL_MS_MIN_VALUE = 5_000;
+
+  public static final String FILE_SAVE_DIR_KEY = "file-save-dir";
+  public static final String FILE_SAVE_DIR_DEFAULT_VALUE = System.getProperty("user.dir");
```
Comment (SteveYurongSu) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/config/ConsumerConstant.java:46]:
```suggestion
  public static final String FILE_SAVE_DIR_DEFAULT_VALUE = System.getProperty("user.dir") + "iotdb-subscription";
```
?

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/config/TopicConfig.java [hunk 2]
```diff
@@ -82,6 +82,15 @@ public Map<String, String> getAttributesWithTimeRange(long creationTime) {
     return attributesWithTimeRange;
   }
 
+  public Map<String, String> getAttributesWithSourceRealtimeMode() {
+    if (TopicConstant.FORMAT_TS_FILE_HANDLER_VALUE.equals(
```
Comment (SteveYurongSu) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/config/TopicConfig.java:86]: return a ? b : c;

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionIncompatibleHandlerException.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  @Override
+  public boolean equals(final Object obj) {
+    return obj instanceof SubscriptionIncompatibleHandlerException
+        && Objects.equals(
+            getMessage(), ((SubscriptionIncompatibleHandlerException) obj).getMessage())
+        && Objects.equals(
+            getTimeStamp(), ((SubscriptionIncompatibleHandlerException) obj).getTimeStamp());
+  }
```
Comment (SteveYurongSu) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionIncompatibleHandlerException.java:41]: hashCode

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionParameterNotValidException.java [hunk 4]
```diff
@@ -23,24 +23,19 @@
...
   }
 
   @Override
-  public boolean equals(Object obj) {
+  public boolean equals(final Object obj) {
     return obj instanceof SubscriptionParameterNotValidException
         && Objects.equals(getMessage(), ((SubscriptionParameterNotValidException) obj).getMessage())
         && Objects.equals(
             getTimeStamp(), ((SubscriptionParameterNotValidException) obj).getTimeStamp());
   }
```
Comment (SteveYurongSu) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionParameterNotValidException.java:40]: hashCode

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionIncompatibleHandlerException.java (hunk 3 above)
Comment (VGalaxies) [reply iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionIncompatibleHandlerException.java:41]: Method 'hashCode()' is identical to its super method

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionParameterNotValidException.java (hunk 4 above)
Comment (VGalaxies) [reply iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionParameterNotValidException.java:40]: Method 'hashCode()' is identical to its super method

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionConnectionException.java [hunk 5]
```diff
@@ -21,25 +21,20 @@
...
-        && Objects.equals(getTimeStamp(), ((SubscriptionPollTimeOutException) obj).getTimeStamp());
-  }
-
-  @Override
-  public int hashCode() {
-    return Objects.hash(getMessage(), getTimeStamp());
+  public boolean equals(final Object obj) {
+    return obj instanceof SubscriptionConnectionException
+        && Objects.equals(getMessage(), ((SubscriptionConnectionException) obj).getMessage())
+        && Objects.equals(getTimeStamp(), ((SubscriptionConnectionException) obj).getTimeStamp());
```
[the remark this answers is not in the document]
Comment (VGalaxies) [reply iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/subscription/exception/SubscriptionConnectionException.java:38]: Method 'hashCode()' is identical to its super method

**Review by SteveYurongSu — APPROVED** [review]
AMAZING! Brilliant work :_)

**Pull request merged**